src/dagmc/DagMC.cpp

```diff
@@ -39,36 +39,46 @@ const bool counting = false; /* controls counts of ray casts and pt_in_vols */
 const std::map<std::string, std::string> DagMC::no_synonyms;
 
 // DagMC Constructor
-DagMC::DagMC(Interface* mb_impl, double overlap_tolerance, double p_numerical_precision) {
+DagMC::DagMC(std::shared_ptr<moab::Interface> mb_impl, double overlap_tolerance, double p_numerical_precision) {
   moab_instance_created = false;
   // if we arent handed a moab instance create one
-  if (NULL == mb_impl) {
-    mb_impl = new moab::Core();
+  if (nullptr == mb_impl) {
+    mb_impl = std::make_shared<Core>();
     moab_instance_created = true;
   }
 
+  MBI_shared_ptr = std::shared_ptr<Interface>(mb_impl);
```

Comment:
These should be equivalent.
```suggestion
  MBI_shared_ptr = mb_impl;
```